lib/rmt.toit

```diff
@@ -931,6 +934,49 @@ class Out extends Channel_:
         break
       state_.wait-for-state WRITE-STATE_
 
+/**
+Synchronizes the output of multiple $Out channels.
+
+Not all hardware supports this feature. The ESP32 does not, but the ESP32C3, ESP32C6,
+  ESP32S2, and ESP32S3 do.
+*/
+class SyncManager:
+  resource_ /ByteArray? := ?
+
+  /**
+  Creates a sync manager for the given $channels.
+
+  The manager ensures that all channels start writing at the same time. It waits until
+    all channels have been instructed to $Out.write before actually emitting any signal.
+  */
+  constructor channels/List:
+    if channels.size < 2: throw "INVALID_ARGUMENT"
+
+    array := Array_ channels.size
+    channels.size.repeat:
+      channel := channels[it]
+      if channel is not Out: throw "INVALID_ARGUMENT"
+      array[it] = channel.resource_
+    resource_ = rmt_sync_manager_new_ resource-freeing-module_ array
+
+    add-finalizer this:: close
+
+  /**
+  Closes the sync manager.
+  */
+  close -> none:
+    if not resource_: return
+    critical-do:
```

Comment:
Why does this need to be in a critical section? We don't check for cancelation when calling primitives and `remove-finalizer` isn't a monitor operation.

Reply:
copy paste... (and I wasn't sure)
done.